src/mono/mono/mini/mini-amd64.c

```diff
@@ -8244,6 +8261,31 @@ mono_arch_emit_exceptions (MonoCompile *cfg)
 			remove = TRUE;
 			break;
 		}
+		case MONO_PATCH_INFO_X128: {
+			guint8 *pos, *patch_pos;
+			guint32 target_pos;
+
+			/* The SSE opcodes require a 16 byte alignment */
+			code = (guint8*)ALIGN_TO (code, 16);
+
+			pos = cfg->native_code + patch_info->ip.i;
+			if (IS_REX (pos [0])) {
+				patch_pos = pos + 4;
+				target_pos = GPTRDIFF_TO_UINT32 (code - pos - 8);
+			}
+			else {
+				patch_pos = pos + 3;
+				target_pos = GPTRDIFF_TO_UINT32 (code - pos - 7);
+			}
+
+			memcpy (code, patch_info->data.target, 16);
+			code += 16;
+
+			*(guint32*)(patch_pos) = target_pos;
+
+			remove = TRUE;
+			break;
+		}
```

Comment:
The original handling I added was the cause of the failures.

MonoJIT doesn't support the VEX encoding today (only the legacy encoding) and the `R4/R8` patch point logic is effectively hardcoded to `movss/movsd`.

For that scenario, we have:
* 1-byte prefix of `0xF3` (float) or `0xF2` (double)
* Optional REX prefix of `0x44` if using extended registers (XMM8-XMM15)
* 2-byte opcode of `0x0F 0x010`
* 1-byte modR/M byte
* 4-byte RIP relative offset

So this the `R4/R8` logic was getting the offset of the RIP relative offset so it could be patched.

--------------

For `X128` we currently hardcode this to `movups` and so we instead have:
* Optional REX prefix of `0x44` if using extended registers (XMM8-XMM15)
* 2-byte opcode of `0x0F 0x010`
* 1-byte modR/M byte
* 4-byte RIP relative offset

Thus it's 1-byte smaller. We could but don't want to emit `movupd` or `movdqu` as that's 1-byte larger for no benefit.

We could emit `movaps` instead, but there isn't any benefit to that on modern machines (anything since ~2011) and it strictly requires that the address be aligned, which is overall less flexible.

When the VEX support is added this all becomes a constant `4-bytes` before the RIP relative offset is encoded (3-byte VEX encoded opcode + modR/M byte).